File: 0201-0300/0201.Bitwise_AND_of_Numbers_Range.cpp
//3ms






class Solution {
public:
    int rangeBitwiseAnd(int left, int right) {
        int shifts = 0;

        while (left != right) {
            left >>= 1;
            right >>= 1;
            shifts++;
        }

        return left << shifts;
    }
};





//0ms





class Solution {
public:
    int rangeBitwiseAnd(int left, int right) {
        int shift = 0;
        while (left < right) {
            left >>= 1;
            right >>= 1;
            shift++;
        }
        return left << shift;
    }
};

File: 0201-0300/0201.Bitwise_AND_of_Numbers_Range.js
//6ms





/**
 * @param {number} left
 * @param {number} right
 * @return {number}
 */
var rangeBitwiseAnd = function(left, right) {
    // Calculate the common leftmost bits between left and right
    let shift = 0;
    while (left < right) {
        // Right shift both left and right to find common bits
        left >>= 1;
        right >>= 1;
        shift++;
    }
    // Left shift the common bits to the original position
    return left << shift;
};





//3ms






/**
 * @param {number} left
 * @param {number} right
 * @return {number}
 */
var rangeBitwiseAnd = function(left, right) {
    let shift_count = 0
    while (left != right) {
        left = left >> 1
        right = right >> 1
        shift_count++
    }
    return left << shift_count
};
